src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1125,80 +1125,86 @@ public void downloadMailArchiveZipFile(Environment env, HttpServletRequest reque
-							+ site_id + " " + e.getMessage();
-					Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-					.entity(errorMessage).type(MediaType.TEXT_PLAIN)
-					.build();
-					log.error(errorMessage);
-					downloadStatus.put("status", errorMessage);
-
-				} catch (Exception e) {
-					String errorMessage = Utils.STATUS_FAILURE + " Migration status for " + site_id + " "
-							+ e.getClass().getName();
-					Response.status(Response.Status.INTERNAL_SERVER_ERROR)
-					.entity(errorMessage).type(MediaType.TEXT_PLAIN)
-					.build();
-					log.error("downloadMailArchiveZipFile ", e);
-					downloadStatus.put("status", errorMessage);
-				}
-			} else {
-				String userError = "Cannot become user " + userId;
-				log.error(userError);
-				downloadStatus.put("status", userError);
-			}
-
-			// update the status and end_time of migration record
-			setMigrationEndTimeAndStatus(migrationId, repository, downloadStatus);
-
-			return;
-
+            // hold download status
+
+            JSONObject downloadStatus = new JSONObject();
+
+            // login to CTools and get sessionId
+            if (sessionAttributes.containsKey(Utils.SESSION_ID)) {
+                String sessionId = (String) sessionAttributes.get(Utils.SESSION_ID);
+                HttpContext httpContext = (HttpContext) sessionAttributes
+                        .get("httpContext");
+                Map<String, String[]> parameterMap = request.getParameterMap();
+                String destination_type = parameterMap.get("destination_type")[0];
+                try {
+                    //
+                    // Sends the response back to the user / browser. The
+                    // content for zip file type is "application/zip". We
+                    // also set the content disposition as attachment for
+                    // the browser to show a dialog that will let user
+                    // choose what action will he do to the sent content.
+                    //
+                    response.setContentType(Utils.MIME_TYPE_ZIP);
+                    String zipFileName=null;
+                    if(destination_type.equals("mailarchivezip")){
+                        zipFileName = site_id + "_mailarchive.zip";
+                        log.info("*** This is Mail Archive Zip Download ***");
+                    }else if(destination_type.equals("mailarchivembox")) {
+                        zipFileName = site_id + "_mailarchivembox.zip";
+                        log.info("*** This is Mail Archive (Mbox) Zip Download ***");
+                    }
+                    response.setHeader("Content-Disposition",
+                            "attachment;filename=\"" + zipFileName + "\"");
+
+                    ZipOutputStream out = new ZipOutputStream(
+                            response.getOutputStream());
+                    // set compression level to high
+                    out.setLevel(9);
```

Comment:
The compression level should be a property defaulting to 9.  At high levels of compression the effort to compress may rise a great deal even though the added compression is very slight. It would be good to have this be settable if we need to tune performance.